include/fs/littlefs.h

```diff
@@ -35,7 +36,10 @@ struct fs_littlefs {
 
 	/* These structures are filled automatically at mount. */
 	struct lfs lfs;
-	const struct flash_area *area;
+	union {
+		const struct flash_area *area;
+		bool blk_mounted;
```

Comment:
We should have note that `blk_mounted` or `area` should also be initialized to 0 when lfs is not used as global/static automatically set to 0.

Reply:
This "union construct" should be changed to allow the removal of `#include <storage/flash_map.h>` from `littlefs.h`.

Reply:
^^why? how? Do you mean to use more #ifdef-ry here?

Reply:
a *void field?

Reply:
Yes, a void pointer, e.g. *mp like I proposed higher. This makes handling of block and flash backend more equal.